profiles/common/modules/custom/ecas/includes/EcasDenieAuth.inc

```diff
@@ -0,0 +1,227 @@
+<?php
+
+/**
+ * @file
+ * Contains EcasDenieAuth.
+ */
+
+/**
+ * Contains logic to check and manage authentication denie.
+ */
+class EcasDenieAuth {
```

Comment:
In the class name, it should probably be either "deny" or "denied"? "Denie" is not a word, I think? But we'd need to make sure it's changed everywhere consistently, though. We don't wanna cause problems my trying to fix this. If too much trouble, leave it.